blackjax/kernels.py

```diff
@@ -765,6 +765,148 @@ def step_fn(rng_key: PRNGKey, state):
         return SamplingAlgorithm(init_fn, step_fn)
 
 
+class ghmc:
+
+    init = staticmethod(mcmc.ghmc.init)
+    kernel = staticmethod(mcmc.ghmc.kernel)
+
+    def __new__(  # type: ignore[misc]
+        cls,
+        logprob_fn: Callable,
+        step_size: PyTree,
+        alpha: float,
+        delta: float,
+        *,
+        divergence_threshold: int = 1000,
+        noise_gn: Callable = lambda _: 0.0,
+        logprob_grad_fn: Optional[Callable] = None,
+    ) -> SamplingAlgorithm:
+
+        step = cls.kernel(noise_gn, divergence_threshold)
+
+        def init_fn(rng_key: PRNGKey, position: PyTree):
+            return cls.init(rng_key, position, logprob_fn, logprob_grad_fn)
+
+        def step_fn(rng_key: PRNGKey, state):
+            return step(
+                rng_key, state, logprob_fn, step_size, alpha, delta, logprob_grad_fn
+            )
+
+        return SamplingAlgorithm(init_fn, step_fn)  # type: ignore[arg-type]
+
+
+def meads(
+    algorithm: ghmc,
+    logprob_fn: Callable,
+    num_batch: int,
+    batch_size: int,
+    num_steps: int = 1000,
+    *,
+    divergence_threshold: int = 1000,
+    logprob_grad_fn: Optional[Callable] = None,
+) -> AdaptationAlgorithm:
```

Comment:
A docstring would be great. In particular I have no idea what `num_batch` and `batch_size` mean (if I haven't read the paper).

Reply:
Do the authors give reasonable default values for these parameters?